source/patterns/@aws-solutions-constructs/aws-s3-sqs/test/test.s3-sqs.test.ts

```diff
@@ -0,0 +1,161 @@
+// Test deployment w/ existing s3 Bucket
+// --------------------------------------------------------------
+test('Test deployment w/ existing Bucket', () => {
+    // Stack
+    const stack = new Stack();
+    // Helper declaration
+
+    const [myBucket] = defaults.buildS3Bucket(stack, {});
+    new S3ToSqs(stack, 'test-s3-sqs', {
+        existingBucketObj: myBucket
+    });
+    // Assertion 1
+    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
+});
+
+// --------------------------------------------------------------
+// Pattern deployment w/ bucket block public access override
+// --------------------------------------------------------------
+test('Pattern deployment w/ bucket versioning turned off', () => {
+    const stack = new Stack();
+    const props: S3ToSqsProps = {
+        bucketProps: {
+            blockPublicAccess: {
+                blockPublicAcls: false,
+                blockPublicPolicy: true,
+                ignorePublicAcls: false,
+                restrictPublicBuckets: true
+            }
+        }
+    };
+    new S3ToSqs(stack, 'test-s3-sqs', props);
+    expect(stack).toHaveResource("AWS::S3::Bucket", {
+        PublicAccessBlockConfiguration: {
+            BlockPublicAcls: false,
+            BlockPublicPolicy: true,
+            IgnorePublicAcls: false,
+            RestrictPublicBuckets: true
+        }
+    });
+});
+
+// --------------------------------------------------------------
+// Test deployment w/ specific s3 event types
+// --------------------------------------------------------------
+test('Test deployment w/ s3 event types and filters', () => {
+  // Stack
+  const stack = new Stack();
+  // Helper declaration
+  const filter: s3.NotificationKeyFilter = {
+    prefix: 'the/place',
+    suffix: '*.mp3'
+  };
+  const props: S3ToSqsProps = {
+    deployDeadLetterQueue: true,
+    maxReceiveCount: 0,
+    queueProps: {},
+    s3EventTypes: [s3.EventType.OBJECT_REMOVED],
+    s3EventFilters: [filter]
+  };
+  new S3ToSqs(stack, 'test-s3-sqs', props);
+  // Assertion 1
+  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
+  // Assertion 2
+  expect(stack).toHaveResource("Custom::S3BucketNotifications", {
+    NotificationConfiguration: {
+      QueueConfigurations: [
+        {
+          Events: ['s3:ObjectRemoved:*'],
+          Filter: {
+            Key: {
+              FilterRules: [
+                {
+                  Name: 'suffix',
+                  Value: '*.mp3'
+                },
+                {
+                  Name: 'prefix',
+                  Value: 'the/place'
+                }
+              ]
+            }
+          },
+          QueueArn: {
+            "Fn::GetAtt": [
+              "tests3sqsqueue810CCE19",
+              "Arn"
+            ]
+          }
+        }
+      ]
+    }
+  });
+  // Assertion 1
+  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
+});
```

Comment:
Please add a unit test to assert the S3 is granted required permissions to use the KMS Key